src/lsdb/loaders/hipscat/read_hipscat.py

```diff
@@ -66,6 +73,79 @@ def read_hipscat(
     return loader.load_catalog()
 
 
+def read_hipscat_subset(
+    path: str,
+    catalog_type: Type[Dataset] | None = None,
+    storage_options: dict | None = None,
+    search_filter: AbstractSearch | None = None,
+    order: int | None = None,
+    n_pixels: int | None = None,
+    **kwargs,
+) -> Dataset:
```

Comment:
I'm not sure we need a separate read method for the subset. Could we add the search_filter option to `read_hipscat` instead?